clevertap-core/src/main/java/com/clevertap/android/sdk/response/InAppResponse.java

```diff
@@ -75,62 +91,96 @@ public void processResponse(final JSONObject response, final String stringBody,
             } else {
                 logger.verbose(config.getAccountId(),
                         "controllerManager.getInAppFCManager() is NULL, not Updating InAppFC Limits");
+            }
+
+            // TODO get all types of inapps from response - ss, cs, applaunched - DONE
+            // TODO store the inapps (get the old code and move it to some ***Store class) - DONE
+            // TODO save the SS/CS mode from the json response - DONE
+            //      add onChanged for this SS/CS mode to handle case when switching from SS/CS to CS/SS or
+            //      from none to CS/SS to clear data. - DONE
+            // TODO call EvaluationManager.evaluateOnAppLaunchedServerSide(appLaunchedNotifs) - DONE
 
+            Pair<Boolean, JSONArray> inappStaleList = res.getStaleInApps();
+            if (inappStaleList.getFirst()) {
+                clearStaleInAppImpressions(inappStaleList.getSecond(), impressionStore);
             }
 
-            JSONArray inappNotifs;
-            try {
-                inappNotifs = response.getJSONArray(Constants.INAPP_JSON_RESPONSE_KEY);
-            } catch (JSONException e) {
-                logger.debug(config.getAccountId(), "InApp: In-app key didn't contain a valid JSON array");
-                // process metadata response
-                cleverTapResponse.processResponse(response, stringBody, context);
-                return;
+            Pair<Boolean, JSONArray> legacyInApps = res.getLegacyInApps();
+            if (legacyInApps.getFirst()) {
+                displayInApp(legacyInApps.getSecond());
+            }
+
+            Pair<Boolean, JSONArray> appLaunchInApps = res.getAppLaunchServerSideInApps();
+            if (appLaunchInApps.getFirst()) {
+                handleAppLaunchServerSide(appLaunchInApps.getSecond());
+            }
+
+            Pair<Boolean, JSONArray> csInApps = res.getClientSideInApps();
+            if (csInApps.getFirst()) {
+                inAppStore.storeClientSideInApps(csInApps.getSecond());
+            }
+
+            Pair<Boolean, JSONArray> ssInApps = res.getServerSideInApps();
+            if (ssInApps.getFirst()) {
+                inAppStore.storeServerSideInAppsMetaData(ssInApps.getSecond());
+            }
+
+            InAppResourceProvider inAppResourceProvider = new InAppResourceProvider(context, logger);
+            InAppCleanupStrategy cleanupStrategy = new InAppCleanupStrategyCoroutine(inAppResourceProvider);
+            InAppImagePreloaderStrategy preloadStrategy = new InAppImagePreloaderCoroutine(inAppResourceProvider, logger);
+
+            /*InAppImagePreloaderStrategy preloadStrategy = new InAppImagePreloaderExecutors(
+                    CTExecutorFactory.executorResourceDownloader(),
+                    inAppResourceProvider, logger
+            );*/
+
+            InAppImageRepoImpl impl = new InAppImageRepoImpl(cleanupStrategy, preloadStrategy, inAppAssetStore);
```

Comment:
The code is like this because asset store and other inapps store are lazily initialised. So if i create in constructor it might have a null ref. This is the limitation we might need to fix to properly solve this problem as you mentioned.